GLTFExporter: Fix skinIndex export for InterleavedBufferAttribute

querielo opened the pull request:
This PR fixes an issue in `GLTFExporter` where `skinIndex` (JOINTS_0) attributes were being exported incorrectly if they were stored as an `InterleavedBufferAttribute`.

The issue can cause this exception:

<img width="567" height="173" alt="Screenshot 2025-12-22 at 15 11 07" src="https://github.com/user-attachments/assets/2bcaa732-1e38-4162-805e-4860cc6eedf9" />


When `skinIndex` data needed to be converted to `UNSIGNED_SHORT` (e.g. if the source was `Uint32Array`), the exporter was creating a new `Uint16Array` directly from the underlying `attribute.array`.

For `InterleavedBufferAttribute`, `attribute.array` refers to the entire shared buffer containing data for multiple attributes. This resulted in the exported `skinIndex` accessor containing mixed data from other interleaved attributes, ignoring the stride and offset.

querielo committed:
GLTFExporter: Fix skinIndex attribute conversion

donmccurdy commented on examples/jsm/exporters/GLTFExporter.js:
```diff
@@ -1852,7 +1852,17 @@ class GLTFWriter {
+
+					}
+
+				}
```
To match the 'else if' block below, do you mind moving this loop to a new utility function like `GLTFExporter.Utils.toUint16BufferAttribute( attribute )`? Or the existing utility function could be extended and renamed as:

```javascript
modifiedAttribute = GLTFExporter.Utils.toTypedBufferAttribute( attribute, Uint16Array );
```

donmccurdy reviewed (commented):
One minor comment suggested, otherwise this fix looks good to me -- thanks @querielo!

Mugen87 committed:
Update GLTFExporter.js

Rename `toFloat32BufferAttribute()` to `toTypedBufferAttribute()` and use it for `skinIndex`.
